pkg/signal/trap_linux_test.go

```diff
@@ -0,0 +1,66 @@
+// +build linux
+
+package signal
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTrap(t *testing.T) {
+	sigmap := map[string]os.Signal{
+		"TERM": syscall.SIGTERM,
+		"QUIT": syscall.SIGQUIT,
+		"INT":  os.Interrupt,
+	}
+	for k, v := range sigmap {
+		wd, _ := os.Getwd()
+		testHelperCode := wd + "/testfiles/main.go"
+		cmd := exec.Command("go", "build", "-o", "main", testHelperCode)
```

Comment:
I don't think that is always the case. I just tried it on this branch and I see the file `pkg/signal/main` marked as an untracked file in the git checkout. This should use a temporary directory.